duckduckgo/iOS
# Add fastlane lane for making Ad-Hoc builds

Task/Issue URL: https://app.asana.com/0/1203301625297703/1203917761929378/f
CC: @jaceklyp 

**Description**:
Add 2 fastlane lanes:
* `adhoc` – to make an ad-hoc build. The functionality mimics adhoc_build.sh, i.e. it takes two parameters:
    * `output` – required, the name of the directory to put the resulting ipa and dSYM.zip
    * `suffix` – optional; suffix comes between “DuckDuckGo” and the timestamp in the ipa/dSYM.zip file name.
* `sync_signing_adhoc` – to fetch certificates and provisioning profiles required for making adhoc
  builds (the certificate is the same as for App Store distribution, but profiles are separate).
  This one is called automatically by `adhoc` so does not need to be called separately under normal circumstances.

Additionally, since syncing signing requires logging into Apple Developer Portal and it takes Apple ID and password,
a helper was added that takes @duckduckgo.com email address from local repository git config as Apple ID.
If your git user email is different, you’ll need to pass `username:<user>@duckduckgo.com` as parameter to fastlane scripts.



**Steps to test this PR**:
1. Make sure you’ve installed fastlane following instructions in [Install Fastlane](https://app.asana.com/0/1202500774821704/1203766784006610/f)
2. Run `bundle exec fastlane adhoc output:<directory> suffix:<some-suffix>`
3. Make sure that it builds a working ipa with the expected name.



---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
